torch/csrc/distributed/c10d/TCPStore.hpp

```diff
@@ -18,6 +18,30 @@ struct SocketAddress {
   std::uint16_t port{};
 };
 
+class Counter {
+ public:
+  void update(double val);
+  std::unordered_map<std::string, double> observe() const;
```

Comment:
Sorry for the late comment, since this looks very common, maybe we can move it to a util?